### PR TITLE
Cnam 412 flattening descriptive stat

### PR DESCRIPTION
**Pull request by dsun0720**
this ticket consists of 2 parts:

1.  parse flattening meta file
2. flattening join confidence degree

I created a new concept: confidence degree, it’s in fact a proportion of number of lines of a given column between single table and flat table. This percent value presents the validation of table join in flattening.

In the confidence degree, it includes 2 algorithms : 

- calculate confidence degree according to given column in single table(CIP13, CIP07, CamCode, etc...) , this approach is used to DCIR since data is identified by "facture" in single tables
- group lines by keys(ETA_NUM, RSA_NUM) and then calculate confidence degree, this approach is used to MCO and MCO_CE in which data is grouped by "établissement"

### CODE REVIEW

**Comment by MaryanMorel** [comment]
Same remark regarding the documentation, otherwise seems good to me.

Regarding the module structure, do you think flattening should be an independent submodule or would it be a good idea to move the files into stats / core submodules? @ysebiat @dsun0720

**Comment by dsun0720** [comment]
the documentation is added.
@MaryanMorel
I think flattening should be an independent submodule, because , not like core or drivers, this one is used to check the validation of flattening

**Comment by MaryanMorel** [comment]
Ok then!

Beware, the doc format of several functions / methods such as `plot_flat_table_confidence_degree` is not numpydoc (https://numpydoc.readthedocs.io/en/latest/example.html#example)

File: scalpel/flattening/flat_table.py [hunk 1]
```diff
@@ -38,6 +51,12 @@ def __init__(
     def from_json(
         json_content: Dict, single_tables: Dict[str, SingleTable]
     ) -> "FlatTable":
+        """
+        build flat table from metadata
```
Comment (ysebiat) [inline scalpel/flattening/flat_table.py]: Capital letter & point at the end of the sentence.

File: scalpel/flattening/flat_table_collection.py [hunk 2]
```diff
@@ -64,6 +90,10 @@ def get(self, flat_name: str) -> FlatTable:
             )
 
     def add_flat_table(self, name: str, flat_table: FlatTable) -> "FlatTableCollection":
+        """return a new collection in adding a flat table in it
```
Comment (ysebiat) [inline scalpel/flattening/flat_table_collection.py]: Wrong format.

File: scalpel/flattening/flat_table_collection.py [hunk 3]
```diff
@@ -72,26 +102,60 @@ def __iter__(self):
         return iter(self._tables)
 
     def union(self, other: "FlatTableCollection") -> "FlatTableCollection":
+        """
```
Comment (ysebiat) [inline scalpel/flattening/flat_table_collection.py]: Wrong format.

**Comment by MaryanMorel** [comment]
Seems good to me!

**Pull request merged**